### PR TITLE
Add support for EC in the JWKS

### PR DESCRIPTION
**Pull request by mziolekda**
For obscure historical reasons, we support certificate files for RS256, ES256 and ES512, where it comes to JWKS though, we only support RS256. This is a strange omission and a jarring asymmetry. This PR resolves the discrepancy

### CODE REVIEW

File: sdk/libs-scala/jwt/src/main/scala/com/daml/jwt/JwksVerifier.scala [hunk 1]
```diff
@@ -56,9 +57,23 @@ class JwksVerifier(
...
-    val publicKey = jwk.getPublicKey.asInstanceOf[RSAPublicKey]
-    RSA256Verifier(publicKey, jwtTimestampLeeway)
+    try {
+      val jwk = http.get(keyId)
+      val publicKey = jwk.getPublicKey
+      publicKey match {
+        case rsa: RSAPublicKey => RSA256Verifier(rsa, jwtTimestampLeeway)
+        case ec: ECPublicKey if ec.getParams.getCurve.getField.getFieldSize == 256 =>
+          ECDSAVerifier(Algorithm.ECDSA256(ec, null), jwtTimestampLeeway)
+        case ec: ECPublicKey if ec.getParams.getCurve.getField.getFieldSize == 521 =>
```
Comment (atriantafyllos-da) [inline sdk/libs-scala/jwt/src/main/scala/com/daml/jwt/JwksVerifier.scala:67]:
```suggestion
        case ec: ECPublicKey if ec.getParams.getCurve.getField.getFieldSize == 512 =>
```
typo?

Comment (atriantafyllos-da) [reply sdk/libs-scala/jwt/src/main/scala/com/daml/jwt/JwksVerifier.scala:67]: if so should not this have been caught by the tests?

Comment (mziolekda) [reply sdk/libs-scala/jwt/src/main/scala/com/daml/jwt/JwksVerifier.scala:67]: hehe, no. But you are very attentive
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev/src/Microsoft.IdentityModel.Tokens/JsonWebKeyECTypes.cs#L40

Comment (atriantafyllos-da) [reply sdk/libs-scala/jwt/src/main/scala/com/daml/jwt/JwksVerifier.scala:67]: oh okay

**Review by atriantafyllos-da — APPROVED** [review]

**Pull request merged**